akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingSnapshot.scala

```diff
@@ -100,6 +102,9 @@ private[akka] class ReplayingSnapshot[C, E, S](override val setup: BehaviorSetup
     throw new JournalFailureException(msg, cause)
   }
 
+  @InternalStableApi
+  def onRecoveryStart(@unused context: ActorContext[_]): Unit = ()
```

Comment:
do we need something like that in `RequestingRecoveryPermit` also?

Reply:
Good idea! It's not used at the moment but might be useful if we decide to add some metrics or tracing for the time spent waiting for the permit. I'll add it in this PR.

Reply:
+1

I think that can be useful, and maybe also the total time from starting until replay completed (permit + snapshot + replay events)

Reply:
That's a good point. I think it will be more accurate to include waiting for a permit time into the total recovery time.